### PR TITLE
[Removed] RxMoyaProvider and ReactiveSwiftMoyaProvider

### PR DESCRIPTION
**Pull request by SD10**
- [x] Removes the `RxMoyaProvider` and `ReactiveSwiftMoyaProvider` deprecated in `v9.0`. 
- [X] Changes MoyaProviderRx -> MoyaProviderRxSpec 😉
- [x] Add **Removed** Changelog entry

### CODE REVIEW

File: Tests/MoyaProvider+RxSpec.swift [hunk 1]
```diff
@@ -6,7 +6,7 @@ import OHHTTPStubs
 @testable import Moya
 @testable import RxMoya
 
-final class MoyaProviderRx: QuickSpec {
+final class MoyaProviderRxSpec: QuickSpec {
```
Comment (sunshinejr) [inline Tests/MoyaProvider+RxSpec.swift]: 👌

**Comment by sunshinejr** [comment]
Thanks for the work, @SD10! One thing that I didn't say before is that with removal of reactive providers we don't need additional abstraction over reactive methods (we needed these as we shared code (e.g. `rx.request` and `RxMoyaProvider<>().request`). Would you like to remove the unnecessary abstraction as well? ( e.g. [here](https://github.com/Moya/Moya/blob/master/Sources/RxMoya/MoyaProvider%2BRx.swift#L9) and in ReactiveSwift extensions as well)

**Comment by SD10** [comment]
@sunshinejr I'm a little confused about the refactoring that needs to be done to remove the two abstractions. It requires I prefix all `MoyaProvider` methods with `.rx.base` & `.reactive.base` then.

I tried moving the methods back into `extension Reactive where Base: MoyaProviderType` but have problems regarding capturing `self` on a non-class constrained type.

Feel free to take over if you'd like

**Comment by sunshinejr** [comment]
@SD10 oh, let's just fix these linting issues and we can do the refactor in another PR, no worries!

**Pull request merged**